src/test_linked_list.py

```diff
@@ -92,30 +106,27 @@ def test_list_push_empty(init_value):
     assert test_list.header.value == init_value
 
 
-@pytest.mark.parametrize('init_value', TYPE_TABLE)
-def test_list_push_head_value(init_value):
+def test_list_push_head_value():
     """Test push on a non-empty list. Make sure pushing to head."""
     test_list = LinkedList()
-    test_list.push('test_string')
-    test_list.push(init_value)
-    assert test_list.header.value == init_value
+    test_list.push('initial_node')
+    test_list.push('second_node')
+    assert test_list.header.value == 'second_node'
 
 
-@pytest.mark.parametrize('init_value', TYPE_TABLE)
-def test_list_push_pointer(init_value):
+def test_list_push_pointer():
     """Test push on a non-empty list. Make sure that pointer is initializing correctly."""
     test_list = LinkedList()
-    test_list.push('test_string')
-    test_list.push(init_value)
-    assert test_list.header.pointer.value == 'test_string'
+    test_list.push('initial_node')
+    test_list.push('second_node')
+    assert test_list.header.pointer.value == 'initial_node'
 
 
-@pytest.mark.parametrize('init_list, result', TABLE_LENGTHS)
-def test_push_length(init_list, result):
+def test_push_length():
     """Test length is correct after a push."""
-    test_list = LinkedList(init_list)
-    test_list.push('some_string')
-    assert test_list.size() == len(init_list) + 1
+    test_list = LinkedList(['a', 'b'])
+    test_list.push('c')
+    assert test_list.size() == 3
```

Comment:
Echoing Cris' prior comment: you should probably assert that the list length is correct before you push as well.